velox/exec/HashTable.cpp

```diff
@@ -249,6 +250,8 @@ class ProbeState {
           reinterpret_cast<uint8_t*>(table.table_), bucketOffset_);
       hits_ = simd::toBitMask(tagsInTable_ == wantedTags_) & kFullMask;
     }
+    // Throws here if we have looped through all the buckets in the table.
+    VELOX_FAIL("Have looped through all the buckets in table");
```

Comment:
It should fail here because there must be some empty slots so not all buckets should be probed?

Reply:
We only reach here if we couldn't find an empty slot. This is sort of preventing any bugs in the relevant code path that could lead to infinite loop.